### ISSUE TITLE
Need UpdateAndRollback Feature

### ISSUE THREAD

**Issue by victorrodrigueznadq** [liquibase/liquibase#4948]
### Search first

- [X] I searched and no similar issues were found

### Description

I see that there is an updateTestingRollback method, but that's not quite what I need.  I'm creating automated tests that can run at any time and any number of tests can be running simultaneously.  Ordinarily, I would just have each test create a unique database for itself from scratch, but in my case, this is going to be prohibitively expensive and will add an unacceptable amount of time to our builds.  Therefore, I'm thinking of having a test database that is in a known state in time and then have the tests update the database and roll it back to it's original state.  This will require an update and rollback without releasing the databasechangeloglock in between the update and the rollback so that other tests waiting for the lock, won't sneak in and run their update.  

Is this currently possible?  I believe the answer is no, but if so, how and if not, how hard would it be to implement and could you put it on your roadmap please?

### Steps To Reproduce

I see that there is an updateTestingRollback method, but that's not quite what I need.  I'm creating automated tests that can run at any time and any number of tests can be running simultaneously.  Ordinarily, I would just have each test create a unique database for itself from scratch, but in my case, this is going to be prohibitively expensive and will add an unacceptable amount of time to our builds.  Therefore, I'm thinking of having a test database that is in a known state in time and then have the tests update the database and roll it back to it's original state.  This will require an update and rollback without releasing the databasechangeloglock in between the update and the rollback so that other tests waiting for the lock, won't sneak in and run their update.  

Is this currently possible?  I believe the answer is no, but if so, how and if not, how hard would it be to implement and could you put it on your roadmap please?

### Expected/Desired Behavior

I see that there is an updateTestingRollback method, but that's not quite what I need.  I'm creating automated tests that can run at any time and any number of tests can be running simultaneously.  Ordinarily, I would just have each test create a unique database for itself from scratch, but in my case, this is going to be prohibitively expensive and will add an unacceptable amount of time to our builds.  Therefore, I'm thinking of having a test database that is in a known state in time and then have the tests update the database and roll it back to it's original state.  This will require an update and rollback without releasing the databasechangeloglock in between the update and the rollback so that other tests waiting for the lock, won't sneak in and run their update.  

Is this currently possible?  I believe the answer is no, but if so, how and if not, how hard would it be to implement and could you put it on your roadmap please?

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR (Thank you!)

**Comment by kevin-atx** [comment]
@victorrodrigueznadq - This is an interesting testing use-case. I understand what you're trying to accomplish, but adding locking behavior as you've described would likely break how Liquibase works for other use-cases.

Have you considered creating your own lock table that you manage as part of your test infrastructure - locking the table before the update and unlocking after the rollback? Then, you could make any number of Liquibase calls within your test without having to worry that other tests will interfere with your activities.

**Comment by victorrodrigueznadq** [comment]
@kevin-atx yes, I have considered having my own lock table, and it's what I'm going to end up doing until Liquibase adds something like what I'm suggesting.  It seems, from the name, that updateTestingRollback is intended for the express purpose of testing.  So, I thought others in the community might benefit from the functionality I'm suggesting.   It shouldn't be that difficult.  Instead of calling update and rollback separately, each of which would individually lock the database, allowing for someone to sneak in between method calls, you could have one method that does everything before releasing the lock.  This functionality is indispensible when creating the database and schema from scratch is prohibitively expensive.

**Comment by filipelautert** [comment]
We're currently triaging our backlog to ensure we can focus on the issues that matter most. With a large number of open issues, it becomes difficult to give each one the attention it deserves.

This issue has been closed due to inactivity — there have been no updates for over 30 days. If this is still relevant to you, please feel free to reopen it or create a new issue with updated context. We appreciate your contribution!
